packages/polymorphic/src/Example/Polymorphic.example.tsx

```diff
@@ -74,7 +75,8 @@ export const ExampleInferred = InferredPolymorphic<ExampleProps>(
 export const ExampleInferredDefaultButton = InferredPolymorphic<
   ExampleProps,
   'button'
->(({ as = 'button' as PolymorphicAs, title, ...rest }) => {
+>(props => {
+  const { as = 'button' as PolymorphicAs, title, ...rest } = props;
```

Comment:
I think we have to change useInferredPolymorphic to accept a default component. By setting a default here, we override the href / as logic

Reply:
otherwise every component will need to have
const component = as ? as : href ? 'a' : default
useInferredPolymorphic(component, rest) and then we're just doing the logic for it, so we might as well grab usePolymorphic

Reply:
Ah yeah, good catch. On it